x-pack/filebeat/input/httpjson/request_chain_helper.go

```diff
@@ -85,6 +94,98 @@ func evaluateResponse(expression *valueTpl, data []byte, log *logp.Logger) (bool
 	return result, nil
 }
 
+// fetchValueFromContext evaluates a given expression and returns the appropriate value from context variables if present
+func fetchValueFromContext(trCtx *transformContext, expression string) (string, bool, error) {
+	var val interface{}
+
+	switch keys := strings.Split(expression, "."); keys[0] {
+	case lastResponse:
+		respMap, err := responseToMap(trCtx.lastResponse, true)
+		if err != nil {
+			return "", false, err
+		}
+		val, err = iterateRecursive(respMap, keys[1:], 0)
+		if err != nil {
+			return "", false, err
+		}
+	case firstResponse:
+		// since first response body is already a map, we do not need to transform it
+		respMap, err := responseToMap(trCtx.firstResponse, false)
+		if err != nil {
+			return "", false, err
+		}
+		val, err = iterateRecursive(respMap, keys[1:], 0)
+		if err != nil {
+			return "", false, err
+		}
+	default:
+		return "", false, fmt.Errorf("context value not supported: %q", keys[0])
```

Comment:
Sorry, we can add more context here.
```suggestion
		return "", false, fmt.Errorf("context value not supported: %q in %q", keys[0], expression)
```